src/clue.js

```diff
@@ -2,25 +2,167 @@
+        age: 22,
+        description: 'Billionaire video game designer',
+        image: 'https://66.media.tumblr.com/ee7155882178f73b3781603f0908617c/tumblr_phhxc7EhPJ1w5fh03_540.jpg',
+        color: 'purple'
+    },
+
+    {   firstName: 'Kasandra',
+        lastName: 'Scarlet',
+        occupation: 'Actor',
+        age: 31,
+        description: 'She is an A-list movie star with a dark past',
+        image: 'https://www.radiotimes.com/uploads/images/Original/111967.jpg',
+        color: 'red'
+    },
+
+    {   firstName: 'Eleanor',
+        lastName: 'Peacock',
+        occupation: 'Socialité',
+        age: 36,        
+        description: 'She is from a wealthy family and uses her status and money to earn popularity',
+        image: 'https://metrouk2.files.wordpress.com/2016/07/mrs-peacock.jpg',
+        color: 'blue'
+    },
+
+    {   firstName: 'Jack',
+        lastName: 'Mustard',
+        occupation: 'Retired Football player',
+        age: 62,
+        description: 'He is a former football player who tries to get by on his former glory',
+        image: 'https://static.independent.co.uk/s3fs-public/thumbnails/image/2016/07/04/08/unspecified-3.jpg',
+        color: 'yellow'
+    }
+];
 
 // Rooms Array
 
-const roomsArray = [];
+const roomsArray = [
+    
+    {name: 'Dining Room'},
+    {name: 'Conservatory'},
+    {name: 'Kitchen'},
+    {name: 'Study'},
+    {name: 'Library'},
+    {name: 'Billiard Room'},
+    {name: 'Lounge'},
+    {name: 'Ballroom'},
+    {name: 'Hall'},
+    {name: 'Spa'},
+    {name: 'Living Room'},
+    {name: 'Observatory'},
+    {name: 'Theater'},
+    {name: 'Guest House'},
+    {name: 'Patio'}
+];
 
 // Weapons Array
 
-const weaponsArray = [];
+const weaponsArray = [
+    {
+        name: 'rope',
+        weight: 10
+    },
+
+    {
+        name: 'knife',
+        weight: 8
+    },
+
+    {
+        name: 'candlestick',
+        weight: 2
+    },
+
+    {
+        name: 'dumbbell',
+        weight: 30
+    },
+
+    {
+        name: 'poison',
+        weight: 2
+    },
+
+    {
+        name: 'axe',
+        weight: 15
+    },
+
+    {
+        name: 'bat',
+        weight: 13
+    },
+
+    {
+        name: 'trophy',
+        weight: 25
+    },
+
+    {
+        name: 'pistol',
+        weight: 20
+    }
+];
 
 
 // ITERATION 2
 
-function selectRandom() {}
+function selectRandom(array) {
+    if (array === '') {
+        return undefined
+    }
+    selectRandom.array.forEach(element => {
+        for (let i > 0; i < array.length; i ++)
+        return Math.random()
+    });
+}
+
+function pickMystery() {
+    let myMystery = {suspect:'', weapon:'', room:''};
+
+    for (`${selectRandom[suspectsArray.lastName]}`)
+        console.log(myMystery.suspect);
+
+    for (`${selectRandom[weaponsArray.name]}`) 
+        console.log(myMystery.weapon);
 
-function pickMystery() {}
+    for (`${selectRandom[roomsArray.name]}`)
+        console.log(myMystery.room);
 
+        return myMystery
+}
```

Comment:
Aqui não precisamos usar for, é só chamar o função criada em cima e ir guardando o elemento aleatório que ele pegou. 
Ex: 
function pickMystery() {
  let mystery = {};   // criamos um objeto vazio

  let theSuspect = selectRandom(suspectsArray);  // usamos a função criada antes para pegar o item aleatório de cada parte do mistério
  let theWeapon = selectRandom(weaponsArray);
  let theRoom = selectRandom(roomsArray);

  mystery.suspect = theSuspect;  // colocamos cada item selecionado dentro do objeto
  mystery.weapon = theWeapon;
  mystery.room = theRoom;

  return mystery;
}